drv/transceivers-server/src/main.rs

```diff
@@ -531,46 +580,13 @@ impl idl::InOrderTransceiversImpl for ServerImpl {
 
 impl NotificationHandler for ServerImpl {
     fn current_notification_mask(&self) -> u32 {
-        notifications::TIMER_MASK | notifications::SOCKET_MASK
+        notifications::SOCKET_MASK | notifications::TIMER_MASK
     }
 
     fn handle_notification(&mut self, bits: u32) {
-        if (bits & notifications::SOCKET_MASK) != 0 {
-            // Nothing to do here; we'll handle it in the main loop
-        }
-
-        if (bits & notifications::TIMER_MASK) != 0 {
-            // Check for errors
-            if self.leds_initialized {
-                self.update_leds();
-                let errors = match self.leds.error_summary() {
-                    Ok(errs) => errs,
-                    Err(e) => {
-                        ringbuf_entry!(Trace::LEDReadError(e));
-                        Default::default()
-                    }
-                };
-                if errors != self.led_error {
-                    self.led_error = errors;
-                    ringbuf_entry!(Trace::LEDErrorSummary(errors));
-                }
-            } else {
-                ringbuf_entry!(Trace::LEDUninitialized);
-            }
-
-            // Query module presence and update LEDs accordingly
-            let (status, _) = self.transceivers.get_module_status();
-
-            let modules_present = LogicalPortMask(!status.modprsl);
-            if modules_present != self.modules_present {
-                self.modules_present = modules_present;
-                ringbuf_entry!(Trace::ModulePresenceUpdate(modules_present));
-            }
-
-            self.update_thermal_loop(status);
-
-            let next_deadline = sys_get_timer().now + TIMER_INTERVAL;
-            sys_set_timer(Some(next_deadline), notifications::TIMER_MASK);
+        if (bits & notifications::SOCKET_MASK | notifications::TIMER_MASK) != 0
```

Comment:
I know that this is how it's done in `net`, but upon reading the `multitimers` code, I think that the Right Way™ to do this is to store the timer in the `ServerImpl` and handle notifications in `self.handle_notifications`

```rust
    fn handle_notification(&mut self, bits: u32) {
        // Ignore the SOCKET_MASK bit; we'll handle it in the main loop
        
        // Pass all bits to the timer, which picks the one it cares about
        self.timer.handle_notification(bits);
        for t in self.timer.iter_fired() {
            match t {
                Timers::I2C => {
                    server.handle_i2c_loop();
                }
                Timers::SPI => {
                    server.handle_spi_loop();
                }
            }
        }
    }
```
(removing the timer-handling code from the main `loop { ... }`)

This is more efficient, because it skips the calls to `poll_now()`.